src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -94,12 +96,113 @@ void EventPipe::Initialize()
 #endif
     }
 
-
     {
         CrstHolder _crst(GetLock());
         if (tracingInitialized)
             s_state = EventPipeState::Initialized;
     }
+
+    EnableViaEnvironmentVariables();
+}
+
+//
+// If EventPipe environment variables are specified, parse them and start a session
+// 
+void EventPipe::EnableViaEnvironmentVariables()
+{
+    STANDARD_VM_CONTRACT;
+    if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EnableEventPipe) != 0)
+    {
+        LPWSTR eventpipeConfig = NULL;
+        CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeConfig, &eventpipeConfig);
+        LPCWSTR eventpipeOutputPath = Configuration::GetKnobStringValue(W("EventPipeOutputPath"));
+        uint32_t eventpipeCircularBufferMB = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeCircularMB);
+
+        if (eventpipeOutputPath == NULL)
+        {
+            eventpipeOutputPath = W("trace.nettrace");
+        }
+        auto configuration = XplatEventLoggerConfiguration();
+        auto configToParse = eventpipeConfig;
+        bool enableDefaultConfig = false;
+
+        if (configToParse == nullptr || *configToParse == L'\0')
+        {
+            enableDefaultConfig = true;
+        }
+
+        // Count how many providers there are to parse
+        int cnt = 0;
+        static WCHAR comma = W(',');
+        while(configToParse != nullptr)
+        {
+            cnt += 1;
+            auto end = wcschr(configToParse, comma);
+            if (end == nullptr)
+            {
+                break;
+            }
+            configToParse = end + 1;
+        }
+
+        // Create EventPipeProviderConfiguration and start tracing.
+        EventPipeProviderConfiguration* pProviders = nullptr;
+
+        // If COMPlus_EnableEventPipe is set to 1 but no configuration was specified, enable EventPipe session
+        // with the default provider configurations.
+        if (enableDefaultConfig)
+        {
+            // TODO: Enable SampleProfiler once we can mutate a EventPipe session post-creation.
+            cnt = 2;
+            pProviders = new EventPipeProviderConfiguration[cnt];
+            pProviders[0] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntime"), 0x4c14fccbd, 5, nullptr);
+            pProviders[1] = EventPipeProviderConfiguration(W("Microsoft-Windows-DotNETRuntimePrivate"), 0x4002000b, 5, nullptr);
+        }
+        else
+        {
+            configToParse = eventpipeConfig;
+            pProviders = new EventPipeProviderConfiguration[cnt];
+            int i = 0;
+            while (configToParse != nullptr)
+            {
+                auto end = wcschr(configToParse, comma);
+                configuration.Parse(configToParse);
+
+                // SampleProfiler can't be enabled on startup yet.
+                if (wcscmp(W("Microsoft-DotNETCore-SampleProfiler"), configuration.GetProviderName()) == 0)
```

Comment:
Rather than explicitly coding around it I'd suggest putting a comment that says it isn't yet implemented (and then don't enable this entire code path until it is finished)